pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -573,7 +599,12 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 		}
 	}
 
-	if vmSnapshotProgressing(vmSnapshotCpy) {
+	if vmSnapshotDeadlineExceeded(vmSnapshotCpy) {
+		vmSnapshotCpy.Status.Phase = snapshotv1.Failed
```

Comment:
while unlikely, it's logically feasible that the deadline could be reported as expired, but the content could be ready to use here. 

should this be `if !vmSnapshotReady(vmSnapshotCpy) && vmSnapshotDeadlineExceeded(vmSnapshotCpy)`

or maybe that check for ready belongs in the deadline exceeded function.

Reply:
@davidvossel you are right, yet I think the idea is that even if it might have been squeezed in the timeframe once it reached the deadline the snapshot is not 100% what the user wanted since it took too long for some reason.. I think its like we draw some line which was crossed at the last minute but was crossed. so I dont mind, your call